backend/api/v1/database_service.go

```diff
@@ -273,11 +266,15 @@ func filterProjectDatabasesV2(ctx context.Context, s *store.Store, iamManager *i
 		}
 	}
 
+	policy, err := s.GetProjectPolicy(ctx, &store.GetProjectPolicyMessage{
+		ProjectID: &projectID,
+	})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get project policy for project %q", projectID)
+	}
+
 	expressionDBsFromAllRoles := make(map[string]bool)
 	for _, binding := range policy.Bindings {
-		if binding.Role != api.ProjectQuerier && binding.Role != api.ProjectExporter {
-			continue
-		}
 		permissions, err := iamManager.GetPermissions(ctx, common.FormatRole(binding.Role.String()))
 		if err != nil {
 			return nil, errors.Wrapf(err, "failed to get permissions")
```

Comment:
QQ:
Can `expressionDBs := getDatabasesFromExpression(binding.Condition.Expression)` works for all binding role? @RainbowDashy 
When the role is not ProjectQuerier or ProjectExporter, we expect `len(expressionDBs) == 0`

Reply:
It works, but with some additional overhead with getDatabasesFromExpression. I'm ok with that.